python/utils.py

```diff
@@ -659,9 +659,16 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
 
 def _import(name, globals={}, locals={}, fromlist=[], level=None):
-    """ wrapper around builtin import that keeps track of loaded plugin modules """
+    """
+    Wrapper around builtin import that keeps track of loaded plugin modules and blocks
+    certain unsafe imports
+    """
     if level is None:
         level = 0
+
+    if 'PyQt4' in name:
+        raise ImportError('Cannot import PyQt4 classes - this would cause a crash!')
```

Comment:
I'd reformulate that, it does no longer cause a crash, so this is mostly a historical consideration at this point. Instead I'd point out that one probably wants to use the PyQt5 variant instead (to help those doing their first time development and copy code from gis.se).

Reply:
What do you mean "it no longer causes a crash"? Try importing PyQt4.QtCore in master and you'll definitely get a crash. But I'll add the advise anyway.

Reply:
When this patch is applied, it should no longer crash, or am I wrong?

Reply:
Yep, no crash, but no import either. So it *would* cause a crash if they were imported.

Reply:
Yeah, but QGIS _would_ crash in a lot of other situations as well if there wouldn't be some safety measures in place, still we don't have a counter to show the user how many times we've internally run into a `if ( param == nullptr ) return;` ;)

Reply:
@m-kuhn agreed, but I think here we are protecting from external code that could be just copypasted by beginners.